server/pages/index.pug

```diff
@@ -226,6 +226,43 @@ html(lang=lang, dir=isRTL ? 'rtl' : 'ltr', class=isFluidWidth ? 'is-fluid-width'
 				}
 			})();
 
+		script.
+			(function () {
+				function createUUID( timestamp ){
+					var dt = timestamp;
+					var uuid = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+						var r = ( dt + Math.random()*16)%16 | 0;
+						dt = Math.floor(dt/16);
+						return (c=='x' ? r :(r &0x3|0x8) ).toString( 16 );
+					} );
+					return uuid;
+				}
+
+				try {
+					var pathName = window.location.pathname.split( "/" ).slice( 1, 2 )[ 0 ];
+					if ( pathName === "log-in" ) {
+						if ( ! window.calypsoLoadStartTime ) {
+							window.calypsoLoadStartTime = {};
+						}
+						var timestamp = Date.now();
```

Comment:
Might try `performance.now()` instead here. It's isolated from any clock changes while you're trying to time the event.

Reply:
see https://developers.google.com/web/updates/2012/08/When-milliseconds-are-not-enough-performance-now

Reply:
it's supported everywhere now too https://caniuse.com/#search=performance.now

Reply:
It is supported indeed on current browsers,
but I don't want to discriminate all browsers / make a special case for that, milliseconds is a good enough resolution for me to answer the question at hand.

I don't need an exact time either, an answer that will satisfy me is: For how much % of our users calypso's log-in takes more than 15s ?